pkg/controllers/user/nodesyncer/cordonfieldssyncer.go

```diff
@@ -47,31 +55,79 @@ func (m *NodesSyncer) syncCordonFields(key string, obj *v3.Node) error {
 }
 
 func (d *NodeDrain) drainNode(key string, obj *v3.Node) error {
-	if obj == nil || obj.DeletionTimestamp != nil || obj.Spec.DesiredNodeUnschedulable != "drain" {
+	if obj == nil || obj.DeletionTimestamp != nil || obj.Spec.DesiredNodeUnschedulable == "" {
 		return nil
 	}
-	kubeConfig, err := d.getKubeConfig()
-	if err != nil {
-		return err
+
+	if obj.Spec.DesiredNodeUnschedulable != "drain" && obj.Spec.DesiredNodeUnschedulable != "cancel" {
+		return nil
+	}
+
+	if obj.Spec.DesiredNodeUnschedulable == "drain" {
+		nodeMapLock.Lock()
+		if _, ok := d.nodesToContext[obj.Name]; ok {
```

Comment:
we should defer unlock, otherwise the lock will be still held even though you return on line 70